File: cpp_run_lua/character.lua
--
-- Created by IntelliJ IDEA.
-- User: zhenkai
-- Date: 2020/4/28
-- Time: 14:23
-- To change this template use File | Settings | File Templates.
--

player = Character.new("Hero", 100)
player:setHealth(80)
hp = player:getHealth()
name = player:getName()
print("Character name: "..name..". HP = "..hp)

--player:free()

File: cpp_run_lua/character.cpp
//
// Created by zhenkai on 2020/4/28.
//

#include "character.h"

Character::Character(const char* name, int hp) {
    this->name = name;
    health = hp;
}

void Character::say(const char* text) {
    std::cout << name << ":" << text << std::endl;
}

void Character::heal(Character* character) {
    character->setHealth(100);
}